src/components/common/PopUpModal.tsx

```diff
@@ -46,34 +46,45 @@ interface PopUpModalProps {
   messageType?: MessageType;
   okButton?: ReactNode;
   cancelButton?: ReactNode;
-  onClose?: () => void;
+  onClose: () => void;
 }
 
-export const PopUpModal = (props: PopUpModalProps) => {
+export const PopUpModal = (props: PopUpModalProps): JSX.Element => {
   const modalRef = useRef<HTMLInputElement>(null);
-  if (modalRef.current) {
-    modalRef.current.checked = props.visible;
-  }
 
   const icon = getMessageTypeIcon(props.messageType);
+  const eventKeyDown = (event): (e: any) => void => {
+    if (event.key === 'Escape') {
+      props.onClose();
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('keydown', eventKeyDown);
+    return () => {
+      document.removeEventListener('keydown', eventKeyDown);
+    };
+  }, [props.onClose]);
 
   return (
-    <div className="dsp-">
+    <div className="dsp-"
+    hidden={!props.visible}
+   >
       <div className="popUpModal">
         <input
           type="checkbox"
           id={props.id}
           className="modal-toggle"
           ref={modalRef}
         />
-        <div className="modalContainer">
+        <div className="modalContainer" tabIndex={0}>
           <div className="modalBox">
             <div className="modalTitle">
               <div className="titleTextIcon">
                 {!isNil(props.messageType) && icon}
                 {props.title}
               </div>
-              {props.onClose && (
+              {(props.onClose != null) && (
```

Comment:
this prop is not an option, the condition isn't necessary

Reply:
Done :smile: